src/octoprint/filemanager/analysis.py

```diff
@@ -17,6 +17,8 @@
 
 import octoprint.util.gcodeInterpreter as gcodeInterpreter
 
+#Added to check if object exceeds printing area
+from octoprint.printer.profile import PrinterProfileManager
```

Comment:
Shouldn't actually be needed if you change to the suggested approach of instead tracking the min/max coordinates of the object and then utilize that information in the *frontend* to display the warning instead. Also, including this here breaks the unit tests, which is why your PR is currently marked by Travis as broken.